src/main/java/com/minecolonies/entity/ai/citizen/guard/GuardArrow.java

```diff
@@ -0,0 +1,56 @@
+package com.minecolonies.entity.ai.citizen.guard;
+
+import com.blockout.Log;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.entity.EntityCitizen;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.projectile.EntityTippedArrow;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.World;
+
+/**
+ * Arrow class for arrows shot by guards.
+ */
+public class GuardArrow extends EntityTippedArrow
+{
+    private static final String TAG_COLONY = "colony";
+    private Colony colony;
+
+    public GuardArrow(final World worldin)
+    {
+        super(worldin);
+    }
+
+    public GuardArrow(final World worldIn, final EntityCitizen shooter)
+    {
+        super(worldIn, shooter);
+        this.colony = shooter.getColony();
+    }
+
+    @Override
+    public void writeEntityToNBT(final NBTTagCompound compound)
+    {
+        super.writeEntityToNBT(compound);
+        compound.setInteger(TAG_COLONY, colony.getID());
+    }
+
+    @Override
+    public void readEntityFromNBT(final NBTTagCompound compound)
+    {
+        super.readEntityFromNBT(compound);
+        int colonyID = compound.getInteger(TAG_COLONY);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'colonyID' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)